src/Cardano/Wallet/Api/V2/Types/CurrencyUnit.hs

```diff
@@ -0,0 +1,22 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+module Cardano.Wallet.Api.V2.Types.CurrencyUnit
+    ( CurrencyUnit (..)
+    ) where
+
+import Prelude
+
+import Data.Aeson
+    ( FromJSON (..), ToJSON (..), Value (String) )
+import GHC.Generics
+    ( Generic )
+
+data CurrencyUnit
+    = Lovelace
+    deriving (Eq, Generic, Show)
+
+instance FromJSON CurrencyUnit where
+    parseJSON (String "lovelace") = pure Lovelace
+    parseJSON _ = fail "Invalid currency unit"
+instance ToJSON CurrencyUnit where
+    toJSON Lovelace = String "lovelace"
```

Comment:
Here we encode the "lovelace" string manually, as the `ObjectWithSingleField` option for `sumEncoding` does not work for types with only a single constructor.

Reply:
Thanks! This magic combination seems to work for both sum types and types with just a single constructor. I'll use this as the definition of `defaultSumTypeOptions`.